src/plugin.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  IPolicyValidationPluginBeta1,
+  IPolicyValidationContextBeta1,
+  PolicyValidationPluginReportBeta1,
+} from 'aws-cdk-lib';
+import { Report, processReport } from './report';
+import { exec } from './utils';
+
+export interface CheckovValidatorProps {
+  /**
+     * List of checks to run
+     *
+     * @default - all checks are run
+     */
+  readonly check?: string[];
+
+  /**
+     * List of checks to skip
+     *
+     * @default - no checks are skipped
+     */
+  readonly skipCheck?: string[];
+}
+
+/**
+ * A validation plugin using checkov
+ */
+export class CheckovValidator implements IPolicyValidationPluginBeta1 {
+  public readonly name: string;
+
+  private readonly checkov: string;
+  private readonly check: string[];
+  private readonly skipCheck: string[];
+
+  private templatePaths: string[] = [];
+
+  constructor(props: CheckovValidatorProps = {}) {
+    this.name = 'cdk-validator-checkov';
+
+    this.checkov = 'checkov'; // possible improvement allow Docker usage
+    this.check = props.check ?? [];
+    this.skipCheck = props.skipCheck ?? [];
+  }
+
+  validate(context: IPolicyValidationContextBeta1): PolicyValidationPluginReportBeta1 {
```

Comment:
the function name is a bit confusing as in addition to validation it also executes a side effect, just a small suggestion that might work is to remove the `templatePaths` as a class member and use it as a param